### PR TITLE
Donation Form Queue

### PR DESCRIPTION
**Pull request by awang378**
## Donation Form Queue 

Implemented OngoingDonations collection and CRUD endpoints to handle ongoing donations.

### How to Test
Import this Postman collection:
https://www.getpostman.com/collections/da4540dfa30d6ea2a06c

### Important Changes

- Creates OngoingDonation document, schema, and model (based of Donations)
- Added `GET`, `PUT`, and `DELETE` endpoints to handle OngoingDonations
- Modified `POST` request for Donations to save to Users Donations array and OngoingDonations with transactions using async/await
- Added `favorite` field to Dish Schema
- Modified `uploadImageAzure` to return empty string if no image specified in a request.

### Related PRs

- GTBitsOfGood/umi-feeds-app#95

### Checklist

- [x]  Database schema docs have been updated or are not necessary
- [x]  Code follows design and style guidelines
- [x]  Code is commented with doc blocks
- [x]  Tests have been written and executed or are not necessary
- [x]  Latest code has been rebased from base branch (usually `develop`)
- [x]  Commits follow guidelines (concise, squashed, etc)
- [x]  Github issues have been linked in relevant commits
- [x]  Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### CODE REVIEW

**Comment by awang378** [comment]
There is a merge conflict with the sending push notifications, but I think it should be okay to take that function and merge into the updated user `.then` chain. I guess also merge conflict with how Michael handled empty req.files.image.

File: src/controllers/donationFormController.ts [hunk 1]
```diff
@@ -88,52 +88,88 @@ export const postDonationForm = (req: Request, res: Response) => {
...
+    const session = await mongoose.startSession();
+    session.startTransaction();
 
-            let currentUser:UserDocument;
-            [newDonationForm.imageLink, currentUser] = values; // values is [imagelink, currentuser]
+    try {
+        // Handle image upload and finding specified user with async/await
+        let url = '';
+        if (req.files !== null && req.files !== undefined && req.files.image !== undefined) {
+            url = await uploadImageAzure(req.files.image as UploadedFile);
```
Comment (izou3) [inline src/controllers/donationFormController.ts]: optimization here

File: src/controllers/donationFormController.ts [hunk 2]
```diff
@@ -88,52 +88,88 @@ export const postDonationForm = (req: Request, res: Response) => {
...
+            .then((updatedUser: UserDocument) => {
+                const donationId = updatedUser.donations[updatedUser.donations.length - 1]._id;
+                newDonationForm._id = donationId;
+                return updatedUser;
             });
-        } catch (err) {
-            res.status(400).json({ message: err.message, donationform: {} });
+
+        // UploadImage and Query User simultaneously by creating a promise out of both asynchronous tasks
+        // Promise.all(
```
Comment (izou3) [inline src/controllers/donationFormController.ts]: Use these statements as guide to parallelize the two sequential independant awaits

File: src/controllers/donationFormController.ts [hunk 3]
```diff
@@ -88,52 +88,88 @@ export const postDonationForm = (req: Request, res: Response) => {
...
+            console.log('sending notification');
+            sendBatchNotification(`New Donation from ${currentUser.businessName}`,
+                req.body.description ?? 'Check app for details',
+                tokens);
+        });
+
+        // Adds donation to OngoingDonations
+        const savedOngoing = await OngoingDonation.create([newDonationForm], { session });
+
+        if (savedDonation && savedOngoing) {
```
Comment (izou3) [inline src/controllers/donationFormController.ts]: optimization here. Check slack

**Pull request merged**